topdown/copypropagation/copypropagation.go

```diff
@@ -206,9 +210,7 @@ func (t bindingPlugTransform) plugBindingsVar(pctx *plugContext, v ast.Var) (res
 	// Apply binding list to substitute remaining vars.
 	if v, ok := result.(ast.Var); ok {
 		if b, ok := pctx.bindings[v]; ok {
-			if !pctx.negated || b.v.IsGround() {
-				result = b.v
-			}
+			result = b.v
```

Comment:
This change undoes 5cf2d0c86e9faa04ed678d283101783df2ca4d9c and then makes up for it in the post-processing. I'm wondering if instead it would be better to fix it during the plug step.